libvmi/driver/xen/libxc_wrapper.c

```diff
@@ -0,0 +1,141 @@
+/* The LibVMI Library is an introspection library that simplifies access to
+ * memory in a target virtual machine or in a file containing a dump of
+ * a system's physical memory.  LibVMI is based on the XenAccess Library.
+ *
+ * Author: Tamas K Lengyel <tamas.lengyel@zentific.com>
+ *
+ * This file is part of LibVMI.
+ *
+ * LibVMI is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * LibVMI is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with LibVMI.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <string.h>
+
+#include "xen_private.h"
+
+#define LIBXC_PATHS_TO_TRY 2
+static const char* libxc_paths[LIBXC_PATHS_TO_TRY] = {
+    "/usr/local/lib/libxenctrl.so",
+    "/usr/lib/libxenctrl.so",
+};
```

Comment:
I'm concerned that we will run afoul of package maintainers with this approach..using LIBXC_PATHS_TO_TRY like above basically requires keeping a big list of possible paths for all distributions we want to support.  

Case in point: As-is this excludes Red Hat/Centos using the Citrix maintained Xen RPMs because they install libxenctrl into /usr/lib64. Also, those specific packages actually use versioned files too (libxenctrl.so.4.6 for example) so multiple can be present in parallel.

Any prefix of course is possible per compile-time directives for the xen libs.

I recommend looking at ld.so(8) for dynamic linker search order. Relying on system defined resolution (especially given the flexibility of /etc/ld.so.conf and /etc/ld.so.conf.d) is probably the safest route.